ZConfig/_schema_utils.py

```diff
@@ -308,15 +308,38 @@ def _visit_default(self, name, info):
     del TypeVisitor
 
 
-def load_schema(schema, package, package_file):
+def load_schema(schema,
+                package=False,
+                package_file='component.xml',
+                component=True):
+    """
+    Load the *schema* and return the schema object.
+
+    By default, *schema* is interpreted as a path on disk to a schema
+    file.
+
+    If *package* is set to True, then *schema* must name a Python
+    package, and the file will be loaded from that package, defaulting
+    to the ``component.xml`` file as nemed by *package_file*. In this
+    case, a ``<schema>`` is generated that imports this component.
+
+    If *package* is set to True and *component* is set to False, then
+    no schema tag is added and the file must refer to an actual
+    schema.
+    """
+
     if not package:
+        # A schema file
         schema_reader = argparse.FileType('r')(schema)
-    else:
+        return ZConfig.loader.loadSchemaFile(schema_reader)
```

Comment:
Use a `with` statement maybe to avoid ResourceWarnings from unclosed files?

(Unless `loadSchemaFile` explicitly closes the file object passed to it, which would surprise me, but is not out of the realm of possibility.)

Reply:
Happily enough, `loadSchemaFile` is guaranteed to close the file passed to it. (It turns into `BaseLoader.loadFile` which wraps the file in a `Resource` and uses `try/finally` to close the resource; a resource just delegates to its underlying file.)